### DISCUSSION TITLE
SchemaStore entry with empty `fileMatch` array

### DISCUSSION THREAD

**Discussion by pkalita-lbl** [redhat-developer/vscode-yaml#774] [Q&A]
I am interested in validating a YAML file with the LinkML schema. It is already registered with SchemaStore [here](https://github.com/SchemaStore/schemastore/blob/5009ffacc4c4d73d3241ae6873dc9c8e479d89d6/src/api/json/catalog.json#L1637-L1642), like so:

```json
{
  "name": "LinkML Metamodel",
  "description": "LinkML metamodel file",
  "fileMatch": [],
  "url": "https://w3id.org/linkml/meta.schema.json"
},
```

I understand why that entry has `"fileMatch": []`. There is no path or filename pattern that would indicate that the file adheres to that schema; it could be any YAML file in any location. Therefore I'd be happy to use the "Select JSON Schema" UI affordance to manually choose the LinkML schema, but it is not presented as an option in that list. In fact it seems like all of the SchemaStore entries with `"fileMatch": []` are filtered out of that list. Is that intentional behavior?

**Comment by gorkem** [comment]
Yes, it is intentional. We filter out any schema that does have a filteMatch entry with a `.yml` or `.yaml` extension. In the next release, we will also add schemas that match `file.association` settings that are mapped to YAML language mode.

**Comment by delaskoff** [comment]
Hi @gorkem!
I can see there were several releases but none of them had this `file.association` implemented
This should fix the https://github.com/redhat-developer/vscode-yaml/issues/1130 and also https://github.com/redhat-developer/vscode-yaml/pull/710 doesn't seem to work because of empty `fileMatch`

**Reply by gorkem** [reply]
Unfortunately, I have limited access to the project since I left Red Hat but let's ping @msivasubramaniaan if he can take a look.
